app/experimenter/templates/experiments/detail_ship.html

```diff
@@ -39,74 +39,6 @@
       Return to Sign-Off
     </button>
   </form>
-
-  <!-- Modal -->
-  <div class="modal fade" id="normandyModal" tabindex="-1" role="dialog" aria-labelledby="normandyModalLabel" aria-hidden="true">
-    <div class="modal-dialog modal-lg" role="document">
-      <div class="modal-content">
-
-        <div class="modal-header">
-          <h5 class="modal-title" id="normandyModalLabel">Normandy Recipe</h5>
-          <button type="button" class="close" data-dismiss="modal" aria-label="Close">
-            <span aria-hidden="true">&times;</span>
-          </button>
-        </div>
-
-        <div class="modal-body">
-          <div>
-            <p>Start by exporting the recipe to Normandy:</p>
-            <p>
-              <a
-                href="{{ experiment.delivery_console_experiment_import_url }}"
-                target="_blank"
-                class="btn btn-primary"
-                role="button"
-                id="export-normandy-button"
-              >
-                Export to Normandy
-              </a>
-            </p>
-          </div>
-
-          <form
-            method="post"
-            action="{% url "experiments-normandy-update" slug=experiment.slug %}"
-            id="id-form"
-            class="d-none"
-          >
-            {% csrf_token %}
-            <hr />
-
-            <p>After exporting enter the new recipe ID below:</p>
-
-            <div class="form-group">
-              {% include "experiments/field_input_inline.html" with form=normandy_id_form field=normandy_id_form.normandy_id %}
-              {% include "experiments/field_errors_inline.html" with form=normandy_id_form field=normandy_id_form.normandy_id %}
-            </div>
-
-            <div class="form-group">
-              {% include "experiments/field_input_inline.html" with form=normandy_id_form field=normandy_id_form.other_normandy_ids %}
-              {% include "experiments/field_errors_inline.html" with form=normandy_id_form field=normandy_id_form.other_normandy_ids %}
-            </div>
-
-            <div class="form-group">
-              <button type="submit" class="btn btn-success">Save Recipe ID</button>
-            </div>
-          </form>
-        </div>
-
-        <div class="modal-footer">
-          <div class="flex-fill">
-            <a href="{{ experiment.api_recipe_url }}" target="_blank">Preview Recipe JSON</a>
```

Comment:
Oh actually, can you also add this link in the links section (where the bugzilla links are etc) to the experiment recipe json that only shows up for [SHIP, ACCEPTED, LIVE, COMPLETE] because we'll still need to be able to get to that JSON for other stuff.

I've been using a pattern lately of having a method on the model whenever stuff needs to show up selectively like

```py
def should_have_recipe_link(self):
    return whatever
```

and calling that from the template

Reply:
You could also try writing a parameterized test if you're feeling brave ⚔️

Reply:
I lucked out and the method already exists 😁